jobbergate-core/jobbergate_core/tools/sbatch.py

```diff
@@ -0,0 +1,143 @@
+
+    def __post_init__(self):
+        with check_expressions("Check paths", raise_exc_class=ValueError) as check:
+            check(self.sbatch_path.is_absolute(), "sbatch_path is not an absolute path")
+            check(self.sbatch_path.exists(), "sbatch_path does not exist")
+            check(self.scontrol_path.is_absolute(), "scontrol_path is not an absolute path")
+            check(self.scontrol_path.exists(), "scontrol_path does not exist")
+
+    def run(self, job_script_path: Path) -> int:
+        """Runs sbatch as the user."""
+        command = (
+            self.sbatch_path.as_posix(),
+            job_script_path.as_posix(),
+            "--parsable",
+        )
+
+        completed_process = self._run_command_as_user(
+            command, cwd=self.submission_directory, capture_output=True, text=True
+        )
+
+        try:
+            return int(completed_process.stdout.split(",")[0])
+        except (ValueError, AttributeError) as e:
+            message = f"Failed to parse slurm job id from {completed_process.stdout}"
+            logger.error(message)
+            raise RuntimeError(message) from e
+
+    def get_job_info(self, job_id: int) -> str:
+        """Gets job info as the user."""
+        command = (
+            self.scontrol_path.as_posix(),
+            "show",
+            "job",
+            str(job_id),
+            "--json",
+        )
+        completed_process = self._run_command_as_user(command, capture_output=True, text=True)
+        data = json.loads(completed_process.stdout)
+        try:
+            return json.dumps(data["jobs"][0])
+        except KeyError as e:
+            message = f"Failed to parse job info from {completed_process.stdout}"
+            logger.error(message)
+            raise RuntimeError(message) from e
+        except IndexError as e:
+            message = f"Job not fount: {job_id}"
+            logger.warning(message)
+            raise RuntimeError(message) from e
+
+    def copy_file_to_submission_directory(self, source_file: Path) -> Path:
+        """Copies the job file to the submission directory as the user."""
+        # Reference: https://stackoverflow.com/a/71589233
+        destination_file = self.submission_directory / source_file.name
+        command = ("tee", destination_file.as_posix())
+        try:
+            with source_file.open("rb") as source:
+                self._run_command_as_user(
+                    command,
+                    stdin=source,
+                    stdout=subprocess.DEVNULL,
+                )
+        except IOError as e:
+            message = f"Failed to copy file to submission directory: {e}"
+            logger.error(message)
+            raise RuntimeError(message) from e
+        return destination_file
+
+    def create_submission_directory(self) -> Path:
+        """Creates a submission directory as the user."""
+        command = ("mkdir", "--parents", self.submission_directory.as_posix())
```

Comment:
Double-check here the owner of the new folders. Initial tests found it was root, not the user running the command